third_party/ienn/src/ie_compilation.cpp

```diff
@@ -156,28 +152,28 @@ int32_t Compilation::AddInput(uint32_t index) {
     return result;
   }
   try {
-    std::string name(std::to_string(index));
-    size_t layer_id =
-        builder_->addLayer(Builder::InputLayer(name).setPort(Port(dims)));
-    layer_id_map_[index] = layer_id;
+    auto input_node =
+        std::make_shared<op::Parameter>(element::f32, Shape(dims));
+    index_op_map_[index] = input_node;
+    inputs_.push_back(input_node);
   } catch (const std::exception& ex) {
-    std::cout << "[IE] failed to add input layer " << ex.what() << std::endl;
+    std::cout << "[IE] failed to add input layer " << ex.what();
     return error_t::OP_FAILED;
   }
   return error_t::NOT_ERROR;
 }
 
 int32_t Compilation::AddOutput(uint32_t index) {
-  if (layer_id_map_.find(index) == layer_id_map_.end()) {
+  if (index_op_map_.find(index) == index_op_map_.end()) {
     std::cout << "Layer for output index " << index << " is not ready";
     return error_t::BAD_DATA;
   }
-  const size_t input_layer_id = layer_id_map_[index];
-  std::string name(std::to_string(index));
-  name += std::string("_out");
   try {
-    size_t output_id =
-        builder_->addLayer({{input_layer_id}}, Builder::OutputLayer(name));
+    for (size_t i = 0; i < index_op_map_[index]->get_output_size(); i++) {
```

Comment:
Does it use for argmax two outputs value and index?

Reply:
> Does it use for argmax two outputs value and index?

Yes, need I add comments here?